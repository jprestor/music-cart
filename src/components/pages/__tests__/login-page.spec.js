import React from 'react';
import LoginPage from '../login-page';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const state = {
  login: {
    pending: false,
    success: true,
    error: null,
  },
};

const props = {
  pending: false,
  loginSuccess: true,
  loginError: null,
};

const {
  Component,
  mapStateToProps,
  mapDispatchToProps,
  mockDispatch,
} = LoginPage;

describe('Testing Component', () => {
  const element = shallow(<Component {...props} />).instance();

  it('renders correctly', () => {
    expect(element).toMatchSnapshot();
  });
});

describe('Testing mapStateToProps', () => {
  it('should reading the correct keys from store', () => {
    expect(mapStateToProps(state)).toEqual(props);
  });
});

describe('Testing mapDispatchToProps', () => {
  it('should return login method', () => {
    expect(mapDispatchToProps().login).toBeDefined();
  });
});
