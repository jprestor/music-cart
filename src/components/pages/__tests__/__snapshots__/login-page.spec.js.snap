// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Component renders correctly 1`] = `
LoginPage {
  "context": Object {},
  "onSubmit": [Function],
  "props": Object {
    "loginError": null,
    "loginSuccess": true,
    "pending": false,
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "email": "admin@example.com",
    "password": "admin",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <LoginPage
        loginError={null}
        loginSuccess={true}
        pending={false}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <Redirect
        to="/account/"
      />,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
